### PR TITLE
[AArch64] Guarantees smashableMovq adjustment during relocation.

### PR DESCRIPTION
**Pull request by dave-estes**
smashableMovqs often have an internal reference which needs to be
adjusted. While that adjustment code is present, it's only
triggered when the update internal reference flag is set. This
patch insures that the flag is set whenever a smashableMovq is
relocated.

### CODE REVIEW

**Comment by dave-estes** [comment]
Ping @mxw

File: hphp/runtime/vm/jit/relocation-arm.cpp [hunk 1]
```diff
@@ -590,6 +590,12 @@ size_t relocateImpl(Env& env) {
         // Do nothing, as the instruction word was initially copied above
       }
 
+      // If we just copied the first instruction of a smashableMovq, then it may
+      // with an internal reference that'll need to be adjusted below.
```
Comment (mxw) [inline hphp/runtime/vm/jit/relocation-arm.cpp]: `s/with/have`?

File: hphp/runtime/vm/jit/relocation-arm.cpp [hunk 2]
```diff
@@ -590,6 +590,12 @@ size_t relocateImpl(Env& env) {
...
+      if (!literals.count(src) && isSmashableMovq(srcAddr)) {
+	env.updateInternalRefs = true;
```
Comment (mxw) [inline hphp/runtime/vm/jit/relocation-arm.cpp]: You're off by one space.

File: hphp/runtime/vm/jit/relocation-arm.cpp (hunk 1 above)
Comment (dave-estes) [inline hphp/runtime/vm/jit/relocation-arm.cpp]: Fixed

File: hphp/runtime/vm/jit/relocation-arm.cpp (hunk 2 above)
Comment (dave-estes) [inline hphp/runtime/vm/jit/relocation-arm.cpp]: Sorry

**Pull request closed without merging**